pkg/domain/infra/abi/play.go

```diff
@@ -435,7 +435,7 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 		initContainers = append(initContainers, ctr)
 	}
 	for _, container := range podYAML.Spec.Containers {
-		if !strings.Contains("infra", container.Name) {
+		if !strings.Contains(container.Name, "infra") {
```

Comment:
This looks scary, the sort of thing that gives me the impression that was broken from the first moment and was never tested...?

Reply:
Yes this is indeed scary. I wonder if we should remove this check. Why do we exclude infra by name? A user could have this substring in an actual container name.

Reply:
Would be better to prevent it from being added earlier, when we actually have a Container struct and can check if it's an infra container.

Reply:
We can exclude it but since this was bugged it never excluded it in the first place so I wonder if we should just remove this check.

@cdoern You added this code, any ideas?

Reply:
Ok I tried it and this statement is completely wrong and has to be removed.

It loops over all containers in the k8s yaml, infra containers are never in this file so they do not have to be skipped.
Skipping containers by name is just wrong and excludes actual containers.